python/eos/analysis_file_description.py

```diff
@@ -0,0 +1,309 @@
+
+@dataclass
+class ConstraintPriorDescription(Deserializable):
+     constraint: str
+
+@dataclass
+class PriorComponent(Deserializable):
+    name: str
+    descriptions: list
+
+    @classmethod
+    def from_dict(cls, **kwargs):
+        _kwargs = _copy.deepcopy(kwargs)
+        if "descriptions" in kwargs:
+            if "parameters" in kwargs:
+                eos.error(f'Both \'descriptions\' and \'parameters\' are provided for prior component \'{kwargs["name"]}\', ignoring legacy support for \'parameters\'')
+            _kwargs['descriptions'] = [PriorDescription.from_dict(**d) for d in kwargs['descriptions']]
+        if "parameters" in kwargs:
+            eos.warn(f'\'parameters\' is in the description of prior component \'{kwargs["name"]}\', use \'descriptions\' instead')
+            _kwargs.pop("parameters")
+            _kwargs['descriptions'] = [PriorDescription.from_dict(**d) for d in kwargs['parameters']]
+        return Deserializable.make(cls, **_kwargs)
+
+
+
+@dataclass
+class EOSConstraintDescription(Deserializable):
+    constraint: str
+
+@dataclass
+class ManualConstraintDescription(Deserializable):
+    name: str
+    info: dict
+
+@dataclass
+class PyHFConstraintDescription(Deserializable):
+    file: str
+    parameter_map: dict = field(default_factory=dict)
+
+@dataclass
+class LikelihoodComponent(Deserializable):
+    name: str
+    constraints: list = field(default_factory=list)
+    manual_constraints: list = field(default_factory=list)
+    pyhf: list = field(default_factory=list)
+
+    @classmethod
+    def from_dict(cls, **kwargs):
+        if not ('constraints' in kwargs or 'manual_constraints' in kwargs or 'pyhf' in kwargs):
+            raise ValueError('LikelihoodComponent must have at least one of constraints, manual_constraints, or pyhf')
+        _kwargs = _copy.deepcopy(kwargs)
+        if 'constraints' in kwargs:
+            _kwargs['constraints'] = [EOSConstraintDescription.from_dict(constraint=c) for c in kwargs['constraints']]
+        if 'manual_constraints' in kwargs:
+            _kwargs['manual_constraints'] = [ManualConstraintDescription.from_dict(name=n, info=d) for n, d in kwargs['manual_constraints'].items()]
+        if 'pyhf' in kwargs:
+            _kwargs['pyhf'] = PyHFConstraintDescription.from_dict(**kwargs['pyhf'])
+        return Deserializable.make(cls, **_kwargs)
+
+
+
+@dataclass
+class PosteriorComponent(Deserializable):
+    name: str
+    prior: list
+    likelihood: list
+    global_options: dict = field(default_factory=dict)
+    fixed_parameters: dict = field(default_factory=dict)
+
+
+
+@dataclass
+class ObservableComponent(Deserializable):
+    name: str
+    latex: str
+    unit: str
+    options: dict
+    expression: str
+
+
+
+@dataclass
+class PredictionObservableComponent(Deserializable):
+    name: str
+    kinematics: dict = field(default_factory=dict)
+
+@dataclass
+class PredictionComponent(Deserializable):
```

Comment:
Hm, again, not a component. A full prediction set, similar to what happens for a posterior.

Reply:
Again -  `PredictionDescription` better?

Reply:
Yes.